base/compiler/optimize.jl

```diff
@@ -555,54 +555,93 @@ function convert_to_ircode(ci::CodeInfo, sv::OptimizationState)
             if oldidx < length(labelchangemap)
                 labelchangemap[oldidx + 1] += 1
             end
+            if blockchangemap === nothing
+                blockchangemap = fill(0, length(sv.cfg.blocks))
+            end
+            blockchangemap[block_for_inst(sv.cfg, oldidx)] += 1
             idx += 1
             prevloc = codeloc
         end
-        if code[idx] isa Expr && ssavaluetypes[idx] === Union{}
+        if ssavaluetypes[idx] === Union{} && !(code[idx] isa Core.Const)
```

Comment:
Maybe we want to set up a special and explicit marker type instead of `Const` to indicate a statement is known to be unreachable, e.g.
```julia
struct Unreachable
    stmt
    Unreachable(@nospecialize stmt) = new(stmt)
end